src/pathology-report/ReportComponent.resource.ts

```diff
@@ -1,28 +1,258 @@
+  SAMPLESTATUSUUID,
+  REFERRALSTATUSUUID,
+  SAMPLEDROPOFFUUID,
+  RWINKWAVUHOSPITALUUID,
+  YESCONCEPTUUID,
+} from "../constants.js";
 
 export interface EncounterResult {
+  family_name: string;
+  given_name: string;
+  middle__name: string;
+  patientPhoneNumber: { value: string };
+  patientHealthCenter: string;
+  patientUuid: string;
   uuid: string;
+  encounterId: number;
+  personId: number;
   encounterDatetime: Date;
+  sampleStatusObs: { value: { display: string } };
+  referralStatusObs: { value: { display: string } };
+  sampleDropoffObs: { value: { display: string } };
 }
+export interface Patient {
+  uuid: string;
+}
+export interface Concept {
+  uuid: string;
+  name: string;
+}
+export interface Obs {
+  patientUuid: string;
+  obsDateTime: string;
+  sampleDropoffObsUuid: string;
+  referralStatusObsUuid: string;
+  sampleStatusObsUuid: string;
+  location: string;
+  encounterUuid: string;
+  valueCodedName: string;
+  voided: boolean;
+}
+
+// export async function getPatientEncounters(patientUuid): Promise<Array<EncounterResult>> {
+//   const searchPatientEncounters = await openmrsFetch(
+//     `/ws/rest/v1/encounter?encounterType=${ENCOUNTERTYPEUUID}&patient=${patientUuid}`,
+//   );
+//   const fullPatientEncounterData = searchPatientEncounters.data.results.map(async (searchEnc) => {
+//     const enc = await openmrsFetch(`/ws/rest/v1/encounter/${searchEnc.uuid}?v=full`);
+//     const searchPatient = await openmrsFetch(`/ws/rest/v1/patient/${patientUuid}?v=full`);
+//     const retrievedObject = {
+//       patientName: searchPatient.data.person.names[0],
+//       patientPhoneNumber: searchPatient.data.person.attributes.filter(
+//         (attribute) => attribute.attributeType.uuid == PHONENUMBERATTRUUID,
+//       )[0],
+//       patientHealthCenter: searchPatient.data.person.attributes.filter(
+//         (attribute) => attribute.attributeType.uuid == HEALTHCENTERATTRTYPEUUID,
+//       )[0],
+//       patientUuid: searchPatient.data.uuid,
+//       encUuid: enc.data.uuid,
+//       encounterDatetime: enc.data.encounterDatetime,
+//       sampleStatusObs: enc.data.obs.filter((obs) => obs.concept.uuid === SAMPLESTATUSUUID)[0],
+//       referralStatusObs: enc.data.obs.filter((obs) => obs.concept.uuid === REFERRALSTATUSUUID)[0],
+//       sampleDropoffObs: enc.data.obs.filter((obs) => obs.concept.uuid === SAMPLEDROPOFFUUID)[0],
+//     };
 
-export async function getPatientEncounters(patientUuid): Promise<Array<EncounterResult>> {
-  const searchPatientEncounters = await openmrsFetch(
-    `/ws/rest/v1/encounter?patient=c604dabc-2700-102b-80cb-0017a47871b2`,
+//     return retrievedObject;
+//   });
+//   return Promise.all(fullPatientEncounterData);
+// }
+
+export async function getUserLocation() {
+  const session = await openmrsFetch(`/ws/rest/v1/session`);
+  const personUuid = session.data.user.person.uuid;
+  const person = await openmrsFetch(
+    `/ws/rest/v1/person/${personUuid}/attribute`
```

Comment:
I'm pretty sure that the user location is on the `SessionUser` object provided by [useSessionUser](https://github.com/openmrs/openmrs-esm-core/blob/master/packages/framework/esm-react-utils/docs/API.md#usesessionuser).